### ISSUE TITLE
failed to sufficiently increase receive buffer size

### ISSUE THREAD

**Issue by thb007** [AdguardTeam/dnsproxy#204]
```
Feb 11 22:12:13 dot  dnsproxy[11184]: 2022/02/11 22:12:13 [info] Creating the UDP server socket
Feb 11 22:12:13 dot  dnsproxy[11184]: 2022/02/11 22:12:13 [info] Listening to udp://[::]:5312
Feb 11 22:12:13 dot  dnsproxy[11184]: 2022/02/11 22:12:13 [info] Creating a TCP server socket
Feb 11 22:12:13 dot  dnsproxy[11184]: 2022/02/11 22:12:13 [info] Listening to tcp://[::]:5312
Feb 11 22:12:13 dot  dnsproxy[11184]: 2022/02/11 22:12:13 [info] Creating a QUIC listener
Feb 11 22:12:13 dot  dnsproxy[11184]: 2022/02/11 22:12:13 failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 2048 kiB, got: 416 kiB). See https://github.com/lucas-clemente/quic-go/wiki/UDP-Receive-Buffer-Size for details.
Feb 11 22:12:13 dot  dnsproxy[11184]: 2022/02/11 22:12:13 [info] Listening to quic://[::]:8053
Feb 11 22:12:13 dot  dnsproxy[11184]: 2022/02/11 22:12:13 [info] Entering the UDP listener loop on [::]:5312
Feb 11 22:12:13 dot  dnsproxy[11184]: 2022/02/11 22:12:13 [info] Entering the tcp listener loop on [::]:5312
Feb 11 22:12:13 dot  dnsproxy[11184]: 2022/02/11 22:12:13 [info] Entering the DNS-over-QUIC listener loop on [::]:8053



```

**Comment by ameshkov** [comment]
Not a bug. The issue is explained here:
https://github.com/lucas-clemente/quic-go/wiki/UDP-Receive-Buffer-Size

**Comment by guillaume-uH57J9** [comment]
While large buffer size is better for high-bandwidth server, measurements show larger buffers are worse (ie more packet loss) for servers with low-to-average bandwidth. 
See [Higher rmem_max value leading to more packet loss](https://serverfault.com/questions/410230/higher-rmem-max-value-leading-to-more-packet-loss)

dnsproxy exposes option "--udp-buf-size=", so I tried using this parameter to use my system default: 

> ./dnsproxy --udp-buf-size=0 -u ...

> ./dnsproxy --udp-buf-size=212992 ...


Unfortunately the warning persist regarless of parameter udp-buf-size:  
> dnsproxy[1180629]: 2022/12/03 13:07:08 failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 2048 kiB, got: 416 kiB). See https://github.com/lucas-clemente/quic-go/wiki/UDP-Receive-Buffer-Size for details.

This is confusing, I don't know if the parameter is being ignored, or if there's something else going on. 
To investigate I added logging to proxy/server_udp.go to verify that the configuration is passed to the UDP Listener. 
It shows the requested value is use when calling udpListen.SetReadBuffer(...) so I'm wondering why there's still this warning when
